clang/lib/Driver/ToolChains/Darwin.h

```diff
@@ -290,8 +290,50 @@ class LLVM_LIBRARY_VISIBILITY MachO : public ToolChain {
   /// }
 };
 
+/// Apple specific MachO extensions
+class LLVM_LIBRARY_VISIBILITY AppleMachO : public MachO {
+public:
+  AppleMachO(const Driver &D, const llvm::Triple &Triple,
+             const llvm::opt::ArgList &Args);
+  ~AppleMachO() override;
```

Comment:
I think the dtor needs to be virtual. Ditto for the base class's.

Reply:
Does `Darwin` need to be virtual too?

Reply:
wait, never mind, the base class has it, so all the others are implicitly vritual

https://github.com/llvm/llvm-project/blob/main/clang/include/clang/Driver/Tool.h#L46

Reply:
I think yes... sorry my C++ is a little weak